flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/NonDeterministicTests.scala

```diff
@@ -20,52 +20,164 @@ package org.apache.flink.table.planner.expressions
 
 import java.sql.Time
 import java.time.format.DateTimeFormatter
-import java.time.{LocalDateTime, ZoneId}
+import java.time.{LocalDate, LocalDateTime, ZoneId}
 
+import org.apache.flink.api.common.RuntimeExecutionMode
 import org.apache.flink.api.java.typeutils.RowTypeInfo
+import org.apache.flink.configuration.ExecutionOptions
 import org.apache.flink.table.api._
 import org.apache.flink.table.functions.ScalarFunction
 import org.apache.flink.table.planner.expressions.utils.ExpressionTestBase
 import org.apache.flink.types.Row
 
+import org.junit.Assert.assertEquals
 import org.junit.Test
 
+import scala.collection.mutable
+
 /**
   * Tests that check all non-deterministic functions can be executed.
   */
 class NonDeterministicTests extends ExpressionTestBase {
 
   @Test
-  def testCurrentDate(): Unit = {
+  def testCurrentDateTime(): Unit = {
     testAllApis(
       currentDate().isGreater("1970-01-01".toDate),
-      "currentDate() > '1970-01-01'.toDate",
       "CURRENT_DATE > DATE '1970-01-01'",
       "true")
-  }
 
-  @Test
-  def testCurrentTime(): Unit = {
     testAllApis(
       currentTime().isGreaterOrEqual("00:00:00".toTime),
-      "currentTime() >= '00:00:00'.toTime",
       "CURRENT_TIME >= TIME '00:00:00'",
       "true")
+
+    testAllApis(
+      currentTimestamp().isGreater(
+        "1970-01-01 00:00:00".cast(DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE())),
+      s"CURRENT_TIMESTAMP > ${timestampLtz("1970-01-01 00:00:00")}",
+      "true")
+
+    testSqlApi(s"NOW() > ${timestampLtz("1970-01-01 00:00:00")}",
+      "true")
   }
 
   @Test
-  def testCurrentTimestamp(): Unit = {
-    testAllApis(
-      currentTimestamp().isGreater("1970-01-01 00:00:00".toTimestamp),
-      "currentTimestamp() > '1970-01-01 00:00:00'.toTimestamp",
-      "CURRENT_TIMESTAMP > TIMESTAMP '1970-01-01 00:00:00'",
+  def testCurrentDateTimeInStreamMode(): Unit = {
+    val temporalFunctions = getCodeGenFunctions(List(
+      "CURRENT_DATE",
+      "CURRENT_TIME",
+      "CURRENT_TIMESTAMP",
+      "NOW()",
+      "LOCALTIME",
+      "LOCALTIMESTAMP"))
+    val round1 = evaluateFunctionResult(temporalFunctions)
+    Thread.sleep(1 * 1000L)
+    val round2: List[String] = evaluateFunctionResult(temporalFunctions)
+
+    assertEquals(round1.size, round2.size)
+    round1.zip(round2).zipWithIndex.foreach(r => {
+      // CURRENT_DATE may be same between two records
+      if (r._2 == 0) {
+        assert(r._1._1 <= r._1._2)
+      } else {
+        assert(r._1._1 < r._1._2)
+      }
```

Comment:
What a magic code!!! Please improve the code readability. 

```suggestion
    round1.zip(round2).zipWithIndex.foreach {
      case ((result1: String, result2: String), index: Int) => {
        // CURRENT_DATE may be same between two records
        if (index == 0) {
          assert(result1 <= result2)
        } else {
          assert(result2 < result2)
        }
      }
    }
```